src/components/app/ProfileSharing.js

```diff
@@ -171,23 +187,40 @@ class ProfileSharingCompositeButton extends PureComponent<
     });
   }
 
-  _attemptToShare() {
-    if (this.state.state !== 'local' && this.state.state !== 'error') {
+  _attemptToShare(isSecondShare: boolean = false) {
+    if (
+      ((!isSecondShare && this.state.state !== 'local') ||
+        (isSecondShare && this.state.state !== 'public')) &&
+      this.state.state !== 'error'
+    ) {
       return;
     }
     this._notifyAnalytics();
 
-    const { profile, predictUrl } = this.props;
+    const { profile, predictUrl, profileSharingStatus } = this.props;
+    const { shareNetworkUrls } = this.state;
```

Comment:
When reading the following code, it's not so clear to me that this comes from the state then. So either use `this.state.shareNetworkUrls` everywhere, or name this variable in a way that it's obvious that it's the state value. Or maybe rename the state value to something like `shareNetworkUrlsCheckbox` so that we clearly know where it comes from.